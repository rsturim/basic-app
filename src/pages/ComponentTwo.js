import React, { Component } from 'react';

import Nav from '../components/Nav';
import Header from '../components/Header';
import Address from '../components/Address';

class ComponentTwo extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="container">
                <div>
                    <Header>Address</Header>
                    <Nav item="/two" />

                    <p className="lead">
                        Lorem ipsum dolor sit amet, debitis integer integer
                        viverra, dui mauris feugiat cras facilisis.
                    </p>
                    <hr className="my-4" />
                    <p>
                        Vitaeasdasds sed mi pulvinar, pede nec quisque sed nisl
                        in, sapien nam in tincidunt amet suspendisse vel. Wisi a
                        curabitur. Cum pretium fusce fusce, faucibus etiam
                        ligula, orci pretium ut quam aliquam sociis, arcu
                        luctus. Donec vel lectus quisque vel, libero lorem ac
                        velit quisque bibendum pede, justo imperdiet nullam
                        magna, faucibus euismod dolor.
                    </p>
                    <Address
                        person="Joe Smith"
                        street="123 Cupcake lane"
                        city="Emerald City"
                        state="VT"
                        zip="05445"
                    />
                </div>
            </div>
        );
    }
}

ComponentTwo.propTypes = {};

export default ComponentTwo;
