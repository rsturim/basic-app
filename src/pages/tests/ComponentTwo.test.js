import React from 'react';

import ComponentTwo from '../ComponentTwo';
import { MemoryRouter } from 'react-router-dom';

import { mount, shallow } from 'enzyme';

import toJson from 'enzyme-to-json';

describe('ComponentTwo', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter>
                <ComponentTwo />
            </MemoryRouter>,
        );
    });

    test('should find and verify and item in the DOM', () => {
        expect(wrapper.find('.display-5').text()).toBe('Component two');
    });

    test('should match the snapshot', () => {
        const shallowTree = shallow(<ComponentTwo />);

        expect(toJson(shallowTree)).toMatchSnapshot();
    });
});
