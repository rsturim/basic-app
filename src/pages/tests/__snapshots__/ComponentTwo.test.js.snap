// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTwo should match the snapshot 1`] = `
<div
  className="container"
>
  <div>
    <Header>
      Component two
    </Header>
    <Nav
      item="/two"
    />
    <p
      className="lead"
    >
      Lorem ipsum dolor sit amet, debitis integer integer viverra, dui mauris feugiat cras facilisis.
    </p>
    <hr
      className="my-4"
    />
    <p>
      Vitaeasdasds sed mi pulvinar, pede nec quisque sed nisl in, sapien nam in tincidunt amet suspendisse vel. Wisi a curabitur. Cum pretium fusce fusce, faucibus etiam ligula, orci pretium ut quam aliquam sociis, arcu luctus. Donec vel lectus quisque vel, libero lorem ac velit quisque bibendum pede, justo imperdiet nullam magna, faucibus euismod dolor.
    </p>
    <Address
      city="Emerald City"
      person="Joe Smith"
      state="VT"
      street="123 Cupcake lane"
      zip="05445"
    />
  </div>
</div>
`;
